src/devices/wifi/nm-wifi-ap.c

```diff
@@ -1213,32 +1212,29 @@ nm_wifi_ap_new_fake_from_connection (NMConnection *connection)
 	s_wireless = nm_connection_get_setting_wireless (connection);
 	g_return_val_if_fail (s_wireless != NULL, NULL);
 
-	ssid = nm_setting_wireless_get_ssid (s_wireless);
-	g_return_val_if_fail (ssid != NULL, NULL);
-	g_return_val_if_fail (g_bytes_get_size (ssid) > 0, NULL);
-
 	ap = (NMWifiAP *) g_object_new (NM_TYPE_WIFI_AP, NULL);
 	priv = NM_WIFI_AP_GET_PRIVATE (ap);
 	priv->fake = TRUE;
-	nm_wifi_ap_set_ssid (ap, g_bytes_get_data (ssid, NULL), g_bytes_get_size (ssid));
 
 	// FIXME: bssid too?
 
 	mode = nm_setting_wireless_get_mode (s_wireless);
-	if (mode) {
-		if (!strcmp (mode, "infrastructure"))
-			nm_wifi_ap_set_mode (ap, NM_802_11_MODE_INFRA);
-		else if (!strcmp (mode, "adhoc"))
+	if (!mode || !strcmp (mode, "infrastructure")) {
+		nm_wifi_ap_set_mode (ap, NM_802_11_MODE_INFRA);
+	} else {
+		if (!strcmp (mode, "adhoc"))
```

Comment:
let's do `if-else-if` without increasing the indentation level... this is a classical selection out of multiple values, it's visually nicer, if they are all presented at the same level.

Reply:
Okay.